packages/cluster-ui/src/statementDetails/statementDetails.tsx

```diff
@@ -435,6 +447,15 @@ export class StatementDetails extends React.Component<
 
     const statsByNode = this.props.statement.byNode;
     const totalWorkload = calculateTotalWorkload(statsByNode);
+    const nodes = stats.nodes
+      ? stats.nodes.sort().map(node => node.toString())
+      : [];
+    const regions: string[] = [];
+    stats.nodes.forEach(node => {
+      if (!regions.includes(nodeRegions[node.toString()]))
+        regions.push(nodeRegions[node.toString()]);
```

Comment:
small nit: `nodeRegions[node.toString()]` is being used twice here. Perhaps store it in a variable to prevent two function invocation and dict lookup. Though `set` might be a more suitable data structure here.

Reply:
done